benchmark/c_str-functions/c_str.h

```diff
@@ -17,23 +18,18 @@ std::size_t leadingSpacesCount(const char *) noexcept;
 
 std::size_t c_strLength(const char *s);
 std::size_t c_strLength_natural(const char *s);
+
 int32_t c_strToI(const char *) noexcept;
 int64_t c_strToL(const char *) noexcept;
+int64_t c_strToL128(const char *) noexcept;
 
-inline int compareAtoi(const char *s) {
-    auto
-        from_stdlib = atoi(s),
-        from_zoo = c_strToI(s);
-    if(from_stdlib != from_zoo) { throw 0; }
-    return from_stdlib;
-}
-
-inline int compareAtol(const char *s) {
-    auto
+template<int64_t (*FUN)(const char *s), int64_t (*BASE)(const char *) = atoll>
+inline int64_t compareAtol(const char *s) {
+    int64_t
         from_stdlib = atoll(s),
```

Comment:
should this not be `from_stdlib = BASE(s),` given your change to the template?

Reply:
Very important catch!

Reply:
With cpp 20 we can pass lambdas and fps much easier with auto templating